subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/TestResultSerializer.java

```diff
@@ -66,6 +66,7 @@ private void write(Collection<TestClassResult> results, Encoder encoder) throws
     private void write(TestClassResult classResult, Encoder encoder) throws IOException {
         encoder.writeSmallLong(classResult.getId());
         encoder.writeString(classResult.getClassName());
+        encoder.writeString(classResult.getClassDisplayName());
```

Comment:
⭕️ We should keep a close eye on the performance test results here. Since you default the display name to the class name, you will serialize the same string twice now. If this affects performance we could optimize it by writing a boolean that tells the deserializer that they are the same. That would reduce storage costs in the common case and make it only 1 byte more in the case they are different.